data/transactions/transaction_test.go

```diff
@@ -159,6 +259,69 @@ func TestWellFormedErrors(t *testing.T) {
 			proto:         curProto,
 			expectedError: fmt.Errorf("transaction invalid range (%d--%d)", 105, 100),
 		},
+		{
+			tx: Transaction{
+				Type: protocol.ApplicationCallTx,
+				Header: Header{
+					Sender:     addr1,
+					Fee:        basics.MicroAlgos{Raw: 1000},
+					LastValid:  105,
+					FirstValid: 100,
+				},
+				ApplicationCallTxnFields: ApplicationCallTxnFields{
+					ApplicationID: 0,
+					ApplicationArgs: [][]byte{
+						[]byte("write"),
+					},
+					ExtraProgramPages: 1,
+				},
+			},
+			spec:          specialAddr,
+			proto:         curProto,
```

Comment:
this test will break after protocol upgrade

Reply:
should I remove this?

Reply:
no, just use v27 proto, not current.